tests/Unit/Controllers/MessagesControllerTest.php

```diff
@@ -0,0 +1,585 @@
+        $this->controller->conversation($this->request);
+        $this->assertTrue(Message::whereId($msg->id)->first()->read);
+    }
+
+    /**
+     * @testdox send: withNoTextGiven -> throwsException
+     */
+    public function testSendWithNoTextGivenThrowsException()
+    {
+        $this->expectException(ValidationException::class);
+        $this->controller->send($this->request);
+    }
+
+    /**
+     * @testdox send: withNoUserIdGiven -> throwsException
+     */
+    public function testSendWithNoUserIdGivenThrowsException()
+    {
+        $this->request = $this->request->withParsedBody([
+            'text'  => 'a',
+        ]);
+        $this->expectException(ModelNotFoundException::class);
+        $this->controller->send($this->request);
+    }
+
+    /**
+     * @testdox send: withMyUserIdGiven -> throwsException
+     */
+    public function testSendWithMyUserIdGivenThrowsException()
+    {
+        $this->request = $this->request->withParsedBody([
+            'text'  => 'a',
+        ]);
+        $this->request->attributes->set('user_id', $this->user_a->id);
+        $this->expectException(HttpForbidden::class);
+        $this->controller->send($this->request);
+    }
+
+    /**
+     * @testdox send: withUnknownUserIdGiven -> throwsException
+     */
+    public function testSendWithUnknownUserIdGivenThrowsException()
+    {
+        $this->request = $this->request->withParsedBody([
+            'text'  => 'a',
+        ]);
+        $this->request->attributes->set('user_id', '1234');
+        $this->expectException(ModelNotFoundException::class);
+        $this->controller->send($this->request);
+    }
+
+    /**
+     * @testdox send: withUserAndTextGiven -> savesMessage
+     */
+    public function testSendWithUserAndTextGivenSavesMessage()
+    {
+        $this->request = $this->request->withParsedBody([
+            'text'  => 'a',
+        ]);
+        $this->request->attributes->set('user_id', $this->user_b->id);
+
+        $this->response->expects($this->once())
+            ->method('redirectTo')
+            ->with('http://localhost/messages/' . $this->user_b->id)
+            ->willReturn($this->response);
+
+        $this->controller->send($this->request);
+
+        $msg = Message::whereText('a')->first();
+        $this->assertEquals($this->user_a->id, $msg->user_id);
+        $this->assertEquals($this->user_b->id, $msg->receiver_id);
+        $this->assertFalse($msg->read);
+    }
+
+    /**
+     * @testdox delete: withNoMsgIdGiven -> throwsException
+     */
+    public function testDeleteWithNoMsgIdGivenThrowsException()
+    {
+        $this->expectException(ModelNotFoundException::class);
+        $this->controller->delete($this->request);
+    }
+
+    /**
+     * @testdox delete: tryingToDeleteSomeonesMessage -> throwsException
+     */
+    public function testDeleteTryingToDeleteSomeonesMessageThrowsException()
+    {
+        $this->expectException(HttpForbidden::class);
```

Comment:
I would add the expected exceptions at the end (before the call) but thats optional